packages/flutter/lib/src/gestures/multitap.dart

```diff
@@ -587,3 +587,411 @@ class MultiTapGestureRecognizer extends GestureRecognizer {
+        || _pendingTap != null) {
+      _reset();
+    }
+    _trackTap(event);
+  }
+
+  bool _representsSameSeries(_TapTracker tap, PointerDownEvent event) {
+    return tap.hasElapsedMinTime() // touch screens often detect touches intermittently
+        && tap.hasSameButton(event)
+        && tap.isWithinGlobalTolerance(event, kDoubleTapSlop);
+  }
+
+  void _trackTap(PointerDownEvent event) {
+    _stopSerialTapTimer();
+    if (onSerialTapDown != null) {
+      final SerialTapDownDetails details = SerialTapDownDetails(
+        globalPosition: event.position,
+        localPosition: event.localPosition,
+        kind: getKindForPointer(event.pointer),
+        buttons: event.buttons,
+        count: _completedTaps.length + 1,
+      );
+      invokeCallback<void>('onSerialTapDown', () => onSerialTapDown!(details));
+    }
+    final _TapTracker tracker = _TapTracker(
+      event: event,
+      entry: GestureBinding.instance!.gestureArena.add(event.pointer, this),
+      doubleTapMinTime: kDoubleTapMinTime,
+    );
+    assert(_pendingTap == null);
+    _pendingTap = tracker;
+    tracker.startTrackingPointer(_handleEvent, event.transform);
+  }
+
+  void _handleEvent(PointerEvent event) {
+    assert(_pendingTap != null);
+    assert(_pendingTap!.pointer == event.pointer);
+    final _TapTracker tracker = _pendingTap!;
+    if (event is PointerUpEvent) {
+      _registerTap(event, tracker);
+    } else if (event is PointerMoveEvent) {
+      if (!tracker.isWithinGlobalTolerance(event, kDoubleTapTouchSlop)) {
+        _reset();
+      }
+    } else if (event is PointerCancelEvent) {
+      _reset();
+    }
+  }
+
+  @override
+  void acceptGesture(int pointer) {
+    assert(_pendingTap != null);
+    assert(_pendingTap!.pointer == pointer);
+    _acceptedGestures[pointer] = true;
+  }
+
+  @override
+  void rejectGesture(int pointer) {
+    _acceptedGestures[pointer] = false;
+    _reset();
+  }
+
+  void _rejectPendingTap() {
+    assert(_pendingTap != null);
+    final _TapTracker tracker = _pendingTap!;
+    _pendingTap = null;
+    // Order is important here; the `resolve` call can yield a re-entrant
+    // `reset()`, so we need to check cancel here while we can trust the
+    // length of our _completedTaps list.
+    _checkCancel(_completedTaps.length + 1);
+    if (_acceptedGestures[tracker.pointer] != false) {
+      tracker.entry.resolve(GestureDisposition.rejected);
+    }
+    _stopTrackingPointer(tracker);
+  }
+
+  @override
+  void dispose() {
+    _reset();
+    super.dispose();
+  }
+
+  void _reset() {
+    if (_pendingTap != null) {
+      _rejectPendingTap();
+    }
+    _pendingTap = null;
+    _completedTaps.clear();
+    _acceptedGestures.clear();
+    _stopSerialTapTimer();
+  }
+
+  void _registerTap(PointerUpEvent event, _TapTracker tracker) {
+    assert(tracker == _pendingTap);
+    assert(tracker.pointer == event.pointer);
+    _startSerialTapTimer();
+    if (_acceptedGestures[event.pointer] != true) {
```

Comment:
Maybe assert before this line that `_acceptedGestures[event.pointer]` is not false?